packages/core/core/src/parcel-v3/RpcEventRouter.js

```diff
@@ -0,0 +1,45 @@
+// @flow
+
+export type HandlerFunc<K, T, R> = [K, T, R];
+export type HandlerCallback<T, R> = (data: T) => R | Promise<R>;
+
+export class RpcEventRouter {
+  #routes: Map<string, HandlerCallback<any, any>>;
+
+  constructor() {
+    this.#routes = new Map();
+  }
+
+  on<T: HandlerFunc<any, any, any>>(
+    event: T[0],
+    callback: HandlerCallback<T[1], T[2]>,
+  ) {
+    this.#routes.set(event, callback);
+  }
+
+  callback: any = async (
```

Comment:
is the `any` here needed?

Reply:
Flow wanted this to property to be typed and for some reason it couldn't infer the type from the assignment 🤷